packages/address/src/AddressFormatter.ts

```diff
@@ -0,0 +1,140 @@
+import {Address, FieldName, Country} from './types';
+import {renderLineTemplate, FIELDS_MAPPING} from './utilities';
+import {loadCountry, loadCountries} from './loader';
+
+const FIELD_REGEXP = /({\w+})/g;
+const LINE_DELIMITER = '_';
+const DEFAULT_FORM_LAYOUT =
+  '{firstName}{lastName}_{company}_{address1}_{address2}_{city}_{country}{province}{zip}_{phone}';
+const DEFAULT_SHOW_LAYOUT =
+  '{lastName} {firstName}_{company}_{address1} {address2}_{city} {province} {zip}_{country}_{phone}';
+
+const ORDERED_COUNTRIES_CACHE: {
+  [locale: string]: Country[];
+} = {};
+
+const COUNTRIES_CACHE: {
+  [locale: string]: {
+    [countryCode: string]: Country;
+  };
+} = {};
+
+export default class AddressFormatter {
+  constructor(private locale: string) {
+    this.locale = locale;
+    COUNTRIES_CACHE[this.locale] = {};
+  }
+
+  updateLocale(locale: string) {
+    this.locale = locale;
+    COUNTRIES_CACHE[this.locale] = {};
+  }
+
+  async getCountry(countryCode: string): Promise<Country> {
```

Comment:
I suppose we can't know what all  the country codes are ahead of time?

It would be nice if we could use a type the caches and these functions a bit stronger.